Refactor URI building to enable consistent method overrides

mrkvon opened the pull request:
## Summary

Refactor URI building logic to use existing methods instead of duplicating `router.build()` calls. This enables consistent method replacement at runtime.
Behavior is unchanged.

## Related Issue

No related issue.

## Changes

- Replace duplicated actor URI building logic in `ContextImpl`'s `getKeyPairsFromIdentifier` with existing `getActorUri()` method
- Replace duplicated followers URI building logic in `ContextImpl`'s `sendActivityInternal` with existing `getFollowersUri()` method  
- ~~Fix minor TypeScript type issues~~ I bailed out of the fixes, hoping it would fix the CI issues, unsuccessfully.

## Benefits

- **Consistent method replacement**: URI building now goes through single methods, enabling clean runtime overrides like `ctx.getActorUri = (id) => new URL(decodeURIComponent(id))` without having to override `ctx.getKeyPairsFromIdentifier` as well.
- **Reduced code duplication**: Eliminates redundant `router.build()` calls.

It does not necessarily reduce amount of lines, due to error handling. 🙂 

## Checklist

- [ ] Did you add a changelog entry to the *CHANGES.md*? (not relevant for refactor)
- [ ] Did you write some relevant docs about this change (if it's a new feature)? (not relevant)
- [ ] Did you write a regression test to reproduce the bug (if it's a bug fix)? (not relevant)
- [ ] Did you write some tests for this change (if it's a new feature)? (not relevant)
- [x] Did you run `deno task test-all` on your machine? (yes, but only fixed issues related to my changes)

## Additional Notes

This refactoring particularly benefits a [use case](https://github.com/soap-opera/soap-opera) where Fedify endpoints are used with custom actors on different domains, identified by their URI. Previously, some code paths bypassed the intended URI building methods, making consistent runtime replacement difficult without also overriding protected methods (`getKeyPairsFromIdentifier`).

mrkvon commented:
The failing CI checks with unrelated TypeScript errors are strange. Has this happened before?

dahlia commented:
The CI failure is now fixed! Could you rebase your commits on the current *main*?

mrkvon commented:
> The CI failure is now fixed! Could you rebase your commits on the current _main_?

Rebased. Checks seem to mostly pass. Publish task failed with an Internal Server Error, possibly a registry issue.
